scripts/deploy/common.rb

```diff
@@ -31,3 +31,52 @@ def open_url(url)
     puts "Opening url #{url}"
     `open #{url}`
 end
+
+def delete_git_branch(branch_name)
+    # Ensure we are not on the same branch that we're trying to delete
+    execute("git checkout #{@branch}")
+
+    # Actually delete the branch
+    execute("git push origin --delete #{branch_name}")
+    execute("git branch -D #{branch_name}")
+end
+
+def switch_to_new_branch(branch_name)
+    # Ensure a different version of this branch doesn't already exist.
+    delete_git_branch(branch_name)
+
+    execute_or_fail("git checkout -b #{branch_name}")
```

Comment:
Typically you want to branch off a specific branch, how do we know what that is?

Reply:
TLDR is it's set in the step above, but that's not super clear.

Reply:
updated to explicitly set the upstream branch to the "deploy branch" here